### ISSUE TITLE
Issue marking files watched

### ISSUE THREAD

**Issue by xmodx10** [ShokoAnime/ShokoServer#748]
## VERSION INFORMATION

Server Version: 3.9.0

Desktop Version: 3.9.0

## LOG FILE

```
FIRST ATTEMPT
[2018-07-15 01:09:10:643] Info|BackgroundWorker.OnDoWork => CommandProcessorGeneral.WorkerCommands_DoWork => CommandRequest_UpdateMyListFileStatus.ProcessCommand Processing CommandRequest_UpdateMyListFileStatus: 7E726D51421A9FAD025A9183A4F662FD
[2018-07-15 01:09:10:643] Info|CommandRequest_UpdateMyListFileStatus.ProcessCommand => AniDBHelper.UpdateMyListFileStatus => CommandRequest_AddFileToMyList.ProcessCommand Processing CommandRequest_AddFileToMyList: [Coalgirls]_Ookami_to_Koushinryou_II_02_(1920x1080_Blu-Ray_FLAC)_[CE735972].mkv - 7E726D51421A9FAD025A9183A4F662FD
[2018-07-15 01:09:10:797] Info|ShokoServiceImplementation.ToggleWatchedStatusOnVideo => SVR_VideoLocal.ToggleWatchedStatus => SVR_AnimeSeries.UpdateStats Starting Updating STATS for SERIES 280 (True - True - True)
[2018-07-15 01:09:11:321] Info|CommandRequest_UpdateMyListFileStatus.ProcessCommand => AniDBHelper.UpdateMyListFileStatus => CommandRequest_AddFileToMyList.ProcessCommand Adding file to list: [Coalgirls]_Ookami_to_Koushinryou_II_02_(1920x1080_Blu-Ray_FLAC)_[CE735972].mkv - Not Watched
[2018-07-15 01:09:14:670] Info|BackgroundWorker.OnDoWork => CommandProcessorGeneral.WorkerCommands_DoWork => CommandRequest_UpdateMyListFileStatus.ProcessCommand Updating file list status: [Coalgirls]_Ookami_to_Koushinryou_II_02_(1920x1080_Blu-Ray_FLAC)_[CE735972].mkv --- 7E726D51421A9FAD025A9183A4F662FD - True
[2018-07-15 01:09:15:089] Info|CommandRequest_RefreshAnime.ProcessCommand => SVR_AniDB_Anime.UpdateStatsByAnimeID => SVR_AnimeSeries.UpdateStats Starting Updating STATS for SERIES 280 (True - True - True)

SECOND ATTEMPT
[2018-07-15 01:10:01:485] Info|BackgroundWorker.OnDoWork => CommandProcessorGeneral.WorkerCommands_DoWork => CommandRequest_UpdateMyListFileStatus.ProcessCommand Processing CommandRequest_UpdateMyListFileStatus: 7E726D51421A9FAD025A9183A4F662FD
[2018-07-15 01:10:01:533] Info|ShokoServiceImplementation.ToggleWatchedStatusOnVideo => SVR_VideoLocal.ToggleWatchedStatus => SVR_AnimeSeries.UpdateStats Starting Updating STATS for SERIES 280 (True - True - True)
[2018-07-15 01:10:04:434] Info|BackgroundWorker.OnDoWork => CommandProcessorGeneral.WorkerCommands_DoWork => CommandRequest_UpdateMyListFileStatus.ProcessCommand Updating file list status: [Coalgirls]_Ookami_to_Koushinryou_II_02_(1920x1080_Blu-Ray_FLAC)_[CE735972].mkv --- 7E726D51421A9FAD025A9183A4F662FD - True
```

## DESCRIPTION

After upgrading to the new version I seem to have come across a wired issue. When marking a file watched that has not been watched before it does not seem to work the first time. Second attempt works and then any further status change on the file works without issue.

I can confirm the file is in my AniDB mylist but the logs seem to indicate shoko is trying to add it again?

## STEPS TO REPRODUCE

1. Go to a series that has not been watched before and mark an individual file as watched >>
![image](https://user-images.githubusercontent.com/25097788/42726851-1e68e6ca-87ce-11e8-9965-31a6973411ca.png)

**Comment by da3dsoul** [comment]
Lovely. I think it's purely linked to Desktop, it updates fine in Nakamori. I'll need to look

**Comment by da3dsoul** [comment]
What worked before, and what doesn't work now?

**Comment by hidden4003** [comment]
We just need to move the code from else into an if and always execute the update after the (optional) add call, as now it blocks the re-triggering of update as update is programmed in such a way that AniDB has priority.

Root of the issue tho is not here, this shouldn't happen as this is a new file but it seems that mylist id wasn't saved. 
```
[2018-07-19 18:09:25:358] Info|BackgroundWorker.OnDoWork => CommandProcessorGeneral.WorkerCommands_DoWork => CommandRequest_AddFileToMyList.ProcessCommand Processing CommandRequest_AddFileToMyList: [HorribleSubs] Hataraku Saibou - 02 [720p].mkv - 88E7A5F27581E12B2CA2617042889836
[2018-07-19 18:09:27:949] Info|BackgroundWorker.OnDoWork => CommandProcessorGeneral.WorkerCommands_DoWork => CommandRequest_AddFileToMyList.ProcessCommand Adding file to list: [HorribleSubs] Hataraku Saibou - 02 [720p].mkv - Not Watched
```

**Comment by hidden4003** [comment]
This is fixed in a61604989a73d114e23de4f1f460d0671e1240a8
